File: backend/routes/matches.js
// routes/matches.js
const express = require('express');
module.exports = (io, engine) => {
  const r = express.Router();
  r.get('/',        (req, res) => res.json({ matches: engine.getAllMatches() }));
  r.get('/:id',     (req, res) => {
    const m = engine.getMatch(req.params.id);
    if (!m) return res.status(404).json({ error: 'Not found' });
    res.json(m);
  });
  r.post('/:id/submit', async (req, res) => {
    try {
      const result = await engine.submitScore({ matchId: req.params.id, ...req.body });
      res.json({ ok: true, match: result });
    } catch (e) { res.status(400).json({ ok: false, error: e.message }); }
  });
  return r;
};

File: BETELITE_v3/BETELITE_v3/go_backend/routes/matches.go
package routes

import (
	"github.com/gofiber/fiber/v2"

	"betelite-go/middleware"
	"betelite-go/services"
)

func SetupMatchRoutes(api fiber.Router) {
	matchGroup := api.Group("/matches", middleware.AuthRequired())

	// Get all active matches
	matchGroup.Get("/", func(c *fiber.Ctx) error {
		activeMatches := services.Engine.GetActiveMatches()
		return c.JSON(fiber.Map{"success": true, "matches": activeMatches})
	})

	// Get specific match by ID
	matchGroup.Get("/:id", func(c *fiber.Ctx) error {
		id := c.Params("id")
		match := services.Engine.GetMatch(id)
		if match == nil {
			return c.Status(404).JSON(fiber.Map{"error": "Match not found"})
		}
		return c.JSON(fiber.Map{"success": true, "match": match})
	})
}
